Update to react-transition-group

jjanssen opened the pull request:
With the release of [React 15.5](https://facebook.github.io/react/blog/2017/04/07/react-v15.5.0.html#discontinuing-support-for-react-addons) there are a few add-ons which have been moved to separate repositories. For some it means they are already migrating to its new namespace. 

At the moment `react-fine-uploader` has a peerDependency to the discontinued add-on `react-addons-css-transition-group`, whereas some projects may already have migrated to `react-transition-group` namespace. This PR continues on the `react-transition-group` namespace.

Commit:
Update to react-transition-group

rnicholus commented on package.json:
```diff
@@ -57,8 +57,8 @@
     "react": "15.4.2",
-    "react-addons-css-transition-group": "15.4.2",
     "react-addons-test-utils": "15.4.2",
+    "react-transition-group": "1.2.0",
```
This should _either_ be a `peerDependency`, _or_ `dependency`. But not both.

jjanssen replied:
Sure, I was a bit confused with the double reference at first. I have made it a peerDependency, seems more relevant to manage it in own projects.

rnicholus commented on src/gallery/index.jsx:
```diff
@@ -1,6 +1,6 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types'
-import ReactCssTransitionGroup from 'react-addons-css-transition-group'
+import { CSSTransitionGroup } from 'react-transition-group'
```
Nitpick, but we can avoid all other code changes simply by changing this to:

```js
import { CSSTransitionGroup as ReactCssTransitionGroup } from 'react-transition-group'
```

jjanssen commented:
Thanks for reviewing, I have updated the PR with your feedback. 

When I originally started the PR `react-transition-group` only had a v1 out, which they have now replaced with a refactored v2. If there are dependencies on v2 it would probably more relevant for a major release.

jjanssen commented:
Ah it seems like travis uses the dependencies for the testsuite. Would it be more relevant to add it to the `devDependencies`. Any thoughts on that matter?

Commit:
Add react-transition-group to the devDependencies for testing

JeffreyATW commented:
Would you mind updating this branch? Interested in getting this to work with React 16!

rnicholus committed:
Merge branch 'master' into feature/react-transition-group

rnicholus commented:
Thanks for the reminder @JeffreyATW. Released to npm as v1.0.4.
